### PR TITLE
[SPARK-44728][PYTHON][DOCS]Add examples to approxQuantile docstring

### PR DESCRIPTION
**Pull request by michaelzhan-db**
### What changes were proposed in this pull request?

Added examples to `DataFrame.approxQuantile`. Also updated the expected input description to reflect input types.

### Why are the changes needed?

To improve pyspark documentation

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

DocTest.

### Was this patch authored or co-authored using generative AI tooling?

No

### CODE REVIEW

File: python/pyspark/sql/dataframe.py [hunk 1]
```diff
@@ -4850,6 +4850,40 @@ def approxQuantile(
...
+        >>> quantiles
+        [1.0, 3.0, 5.0]
+
+        Example 2: Calculating quantiles for multiple columns
+
+        >>> data = [(1, 10), (2, 20), (3, 30), (4, 40), (5, 50)]
+        >>> df = spark.createDataFrame(data, ["col1", "col2"])
+        >>> quantiles = df.approxQuantile(["col1", "col2"], [0.0, 0.5, 1.0], 0.05)
+        >>> quantiles
+        [[1.0, 10.0], [3.0, 30.0], [5.0, 50.0]]
```
Comment (zhengruifeng) [inline python/pyspark/sql/dataframe.py:4870]:
```suggestion
        [[1.0, 3.0, 5.0], [10.0, 30.0, 50.0]]
```

File: python/pyspark/sql/dataframe.py [hunk 2]
```diff
@@ -4850,6 +4850,40 @@ def approxQuantile(
...
+        >>> quantiles
+        [1.0, 3.0, 4.0]
+
+        Example 4: Calculating quantiles with different target precision
+
+        >>> data = [(1,), (2,), (3,), (4,), (5,)]
+        >>> df = spark.createDataFrame(data, ["values"])
+        >>> quantiles = df.approxQuantile("values", [0.0, 0.5, 1.0], 0.2)
+        >>> quantiles
+        [1.0, 3.0, 5.0]  # (with a larger error bound)
```
Comment (zhengruifeng) [inline python/pyspark/sql/dataframe.py:4886]:
```suggestion
        >>> quantiles  # (with a large error bound)
        [1.0, 2.0, 5.0]
```

the median should be 3.0 but get 2.0 due to the low precision

File: python/pyspark/sql/dataframe.py [hunk 3]
```diff
@@ -4850,6 +4850,40 @@ def approxQuantile(
...
+
+        Example 3: Handling null values
+
+        >>> data = [(1,), (None,), (3,), (4,), (None,)]
+        >>> df = spark.createDataFrame(data, ["values"])
+        >>> quantiles = df.approxQuantile("values", [0.0, 0.5, 1.0], 0.05)
+        >>> quantiles
+        [1.0, 3.0, 4.0]
+
+        Example 4: Calculating quantiles with different target precision
```
Comment (zhengruifeng) [inline python/pyspark/sql/dataframe.py:4880]:
```suggestion
        Example 4: Calculating quantiles with low precision
```

This example is to illustrate that when the given `relativeError` is high (0.2), the output may have large error.

may need to explain this in the example

**Comment by zhengruifeng** [comment]
To test this docstring, you can run

`python/run-tests --python-executables python3 --testnames 'pyspark.sql.dataframe'`

in your local env

File: python/pyspark/sql/dataframe.py (hunk 3 above)
Comment (michaelzhan-db) [reply python/pyspark/sql/dataframe.py:4880]: Good idea! I'll update the example description

**Comment by zhengruifeng** [comment]
thanks, merged to master

**Comment by zhengruifeng** [comment]
oops, @michaelzhan-db let's don't use the umbrella ticket SPARK-42728, you could create a subtask under it next time :)

**Comment by zhengruifeng** [comment]
the ticket should be SPARK-44949

**Pull request closed without merging**